src/db/collections/block.rs

```diff
@@ -292,49 +295,101 @@ impl MongoDb {
+    /// Gathers milestone payload analytics.
+    pub async fn get_milestone_analytics(
+        &self,
+        start_timestamp: Option<MilestoneTimestamp>,
+        end_timestamp: Option<MilestoneTimestamp>,
+    ) -> Result<BlockAnalyticsResult, Error> {
+        Ok(self
+            .db
+            .collection::<BlockAnalyticsResult>(BlockDocument::COLLECTION)
+            .aggregate(
+                vec![
+                    doc! { "$match": {
+                        "block.payload.kind": "milestone",
+                        "$nor": [
+                            { "block.payload.essence.timestamp": { "$lt": start_timestamp } },
+                            { "block.payload.essence.timestamp": { "$gte": end_timestamp } },
+                        ],
+                    } },
+                    doc! { "$group" : {
+                        "_id": null,
+                        "count": { "$sum": 1 },
+                    }},
+                ],
+                None,
+            )
+            .await?
+            .try_next()
+            .await?
+            .map(bson::from_document)
+            .transpose()?
+            .unwrap_or_default())
     }
 
-    impl MongoDb {
-        /// Gathers transaction analytics.
-        pub async fn get_transaction_analytics(
-            &self,
-            start_milestone: MilestoneIndex,
-            end_milestone: MilestoneIndex,
-        ) -> Result<TransactionAnalyticsResult, Error> {
-            Ok(self
-                .db
-                .collection::<TransactionAnalyticsResult>(BlockDocument::COLLECTION)
-                .aggregate(
-                    vec![
-                        doc! { "$match": {
-                           "block.payload.kind": "transaction",
-                           "metadata.referenced_by_milestone_index": { "$gte": start_milestone, "$lte": end_milestone },
-                        } },
-                        doc! { "$unwind": "$block.payload.essence.outputs" },
-                        doc! { "$group": {
-                            "_id": null,
-                            "count": { "$sum": 1 },
-                            "total_value": { "$sum": { "$toDouble": "$block.payload.essence.outputs.amount" } },
-                            "avg_value": { "$avg": { "$toDouble": "$block.payload.essence.outputs.amount" } },
-                        } },
-                    ],
-                    None,
-                )
+    /// Gathers tagged data payload analytics.
+    pub async fn get_tagged_data_analytics(
+        &self,
+        start_timestamp: Option<MilestoneTimestamp>,
+        end_timestamp: Option<MilestoneTimestamp>,
+    ) -> Result<BlockAnalyticsResult, Error> {
+        let start_index = if let Some(start_timestamp) = start_timestamp {
+            if let Some(index) = self
+                .find_first_milestone(start_timestamp)
                 .await?
-                .try_next()
+                .map(|ts| ts.milestone_index)
```

Comment:
We should probably create a helper for this in the `db` module as part of the `impl MongoDb {...}`.

Now that I think of it, would it be possible to perform this as a step in the MongoDb aggregation pipeline? If so we could create an issue for that.

Reply:
A helper for what exactly?

Reply:
Hmm, thinking more about it, I guess this is fine.